Istanbul SNBAT cause null pointer dereferences

```
panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6e335a] goroutine 168 [running]: github.com/ethereum/go-ethereum/core/types.(*Block).Hash(0x0, 0x0, 0x0, 0x0, 0x0) /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/core/types/block.go:375 +0x4a github.com/ethereum/go-ethereum/consensus/istanbul/backend.(*Backend).Seal.func2(0xc00031b740, 0xc0000ea120, 0xc02b2c42d0, 0xc000333440, 0xc019111080) /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/consensus/istanbul/backend/engine.go:440 +0x13d created by github.com/ethereum/go-ethereum/consensus/istanbul/backend.(*Backend).Seal /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/consensus/istanbul/backend/engine.go:433 +0x4a9
```

--- a/consensus/istanbul/backend/engine.go
+++ b/consensus/istanbul/backend/engine.go
@@ -431,9 +431,10 @@ func (sb *Backend) Seal(chain consensus.ChainReader, block *types.Block, results
 		for {
 			select {
 			case result := <-sb.commitCh:
+				// Somehow, the block `result` coming from commitCh can be null
 				// if the block hash and the hash from channel are the same,
 				// return the result. Otherwise, keep waiting the next hash.
-				if block.Hash() == result.Hash() {
+				if result != nil && block.Hash() == result.Hash() {
 					results <- result
 					return
 				}

